packages/aserto/src/aserto/client/directory/__init__.py

```diff
@@ -0,0 +1,449 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple, TypedDict
+
+import grpc
+from aserto.directory.common.v2 import (
+    Object,
+    ObjectIdentifier,
+    ObjectTypeIdentifier,
+    PaginationRequest,
+    PermissionIdentifier,
+    Relation,
+    RelationIdentifier,
+    RelationTypeIdentifier,
+)
+from aserto.directory.exporter.v2 import ExporterStub
+from aserto.directory.importer.v2 import ImporterStub
+from aserto.directory.reader.v2 import (
+    CheckPermissionRequest,
+    CheckRelationRequest,
+    GetObjectManyRequest,
+    GetObjectRequest,
+    GetObjectsRequest,
+    GetObjectsResponse,
+    GetRelationRequest,
+    GetRelationsRequest,
+    GetRelationsResponse,
+    ReaderStub,
+)
+from aserto.directory.writer.v2 import (
+    DeleteObjectRequest,
+    DeleteRelationRequest,
+    SetObjectRequest,
+    SetRelationRequest,
+    WriterStub,
+)
+
+
+class NotFoundError(Exception):
+    pass
+
+
+class Directory:
+    def __init__(self, *, address: str, api_key: str, tenant_id: str, ca_cert: str) -> None:
+        self._channel = grpc.secure_channel(
+            target=address, credentials=self._channel_credentials(cert=ca_cert)
+        )
+        self._metadata = self._get_metadata(api_key=api_key, tenant_id=tenant_id)
+        self.reader = ReaderStub(self._channel)
+        self.writer = WriterStub(self._channel)
+        self.importer = ImporterStub(self._channel)
+        self.exporter = ExporterStub(self._channel)
+
+    def get_objects(
+        self, object_type: Optional[str] = None, page: Optional[PaginationRequest] = None
+    ) -> GetObjectsResponse:
+        """Retrieve a page of directory objects by object type and page size.
+        Returns a list of directory objects and a token for the next page if it exists.
+
+        Parameters
+        ----
+        object_type : str
+            a directory object type
+        page : dict(size: int, token: str)
+            paging information — the size of the page, and the pagination
+            start token
+
+        Returns
```

Comment:
I think this section is a little misleading because it makes it seem like the function returns a `(results, page)` tuple when in fact it returns a `GetObjectResponse`.